app/assets/src/components/Samples.jsx

```diff
@@ -1870,12 +1873,16 @@ function SampleCardCheckboxes({
             key={`sample_${sample.db_sample.id}`}
             data-sample-id={sample.db_sample.id}
             className="filled-in checkbox"
-            checked={
-              parent.state.selectedSampleIds.indexOf(sample.db_sample.id) >= 0
-            }
+            checked={checked}
             disabled={report_ready != 1}
           />{" "}
-          <label htmlFor={i}>{sample_name_info}</label>
+          <label
+            data-checked={checked}
+            onClick={parent.selectSample}
+            data-sample-id={sample.db_sample.id}
+          >
+            {sample_name_info}
```

Comment:
Can you wrap `{sample_name_info}` in `<div onClick={e => e.stopPropagation()} ></div>`?

I checked and this seems to work (though you should check as well). it prevents `selectSample` from being called when you click on the label.

The code works right now, but what's actually happening right now is that `parent.selectSample` IS getting called when you click the label. But sampleId resolves to `NaN` and `e.target.getAttribute("data-checked")` is `null`, so the code happens to do what you want. Better to not call `selectSample` though.